### ISSUE TITLE
[Bug]: Failure when using `upgrade`

### ISSUE THREAD

**Issue by RichiCoder1** [storybookjs/storybook#21577]
### Describe the bug

When running `storybook upgrade --prerelease`, it isn't picking up the RC packages and I'm getting the following bizzare messages in the stdout:

```
storybook upgrade --package-manager npm --prerelease --debug    
 • Checking for latest versions of '@storybook/*' packagesinfo ,,| was unexpected at this time.
info
info ,,| was unexpected at this time.
info
 • Installing upgrades • Preparing to install dependencies. ✓
```

### System

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (12) x64 AMD Ryzen 5 5600G with Radeon Graphics
  Binaries:
    Node: 18.15.0 - ~\AppData\Local\nvs\node\18.15.0\x64\node.EXE
    Yarn: 1.22.19 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 9.5.0 - ~\AppData\Local\nvs\node\18.15.0\x64\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.1344.0), Chromium (110.0.1587.63)
  npmPackages:
    @storybook/addon-a11y: ^6.5.16 => 6.5.16
    @storybook/addon-actions: ^6.5.16 => 6.5.16
    @storybook/addon-essentials: ^6.5.16 => 6.5.16
    @storybook/addon-interactions: ^6.5.16 => 6.5.16
    @storybook/addon-links: ^6.5.16 => 6.5.16
    @storybook/builder-webpack5: ^6.5.16 => 6.5.16
    @storybook/manager-webpack5: ^6.5.16 => 6.5.16
    @storybook/react: ^6.5.16 => 6.5.16
    @storybook/test-runner: ^0.9.4 => 0.9.4
    @storybook/testing-library: ^0.0.13 => 0.0.13
```

**Comment by RichiCoder1** [comment]
FWIW, I'm almost certain this is some windows shell issue. Seems like `npm-check-upgrades` is choking on the regex and just falling back to grabbing `latest`. Not sure why.

**Comment by shilman** [comment]
Fixed in #21583
